cmd/snap-bootstrap/README.md

```diff
@@ -12,10 +14,64 @@ _snap-bootstrap_ is the main executable that is run during the initramfs stage o
 1. _snap-bootstrap_ on kernel and base snap upgrades will also handle updating bootloader environment variables to implement A/B or try-boot functionality.
 1. _snap-bootstrap_ then finally may do some additional setup of the root filesystem such as copying some default files for ephemeral system modes such as recover. 
 
-## Degraded recover mode
+## In depth walkthrough
+
+_snap-bootstrap_ operates differently depending on snapd_recovery_mode, so each mode is considered separately below.
+
+Note however that snap-bootstrap, while containing the largest chunk of the logic for the initramfs, is not alone; there is other machinery at work in the rest of the initramfs that takes over after snap-bootstrap has exited successfully. Thus, there are more steps to fully complete the initramfs operations beyond these steps, and ideally at some point the rest of those steps will be moved into snap-bootstrap so they can be more fully tested and documented, but for now, just see the unit files in the initrd for "the-modeenv" and "the-tool" to follow what happens after snap-bootstrap is done.
+
+Additionally, note that in all modes where encryption is enabled, we may open/unlock the TPM keys (or other security hardware in the case of custom ARM fde-hooks for example) in order to generally do encryption related things and regardless of result, we must _always_ re-seal and re-lock the TPM (or other security hardware) before exiting snap-bootstrap to protect the sealed-keys. This is implemented specifically with `secboot.LockSealedKeys`.
```

Comment:
re-seal for the keys is confusing in this context. As resealing keys means something else. 

We must always lock access to the keys


also we do this I think always when there is a TPM etc even is encryption is not enabled